app/main.py

```diff
@@ -924,6 +924,22 @@ def get_available_uberonids(query):
     return jsonify(result)
 
 
+@app.route("/simulation_ui_file/<identifier>")
+def simulation_ui_file(identifier):
+    results = process_results(dataset_search(create_pennsieve_identifier_query(identifier)))
+    results_json = json.loads(results)
+
+    try:
+        item = results_json["results"][0]
+        uri = item["s3uri"]
+        path = item["abi-simulation-file"][0]["dataset"]["path"]
+        key = f"{uri}files/{path}".replace(f"s3://{Config.S3_BUCKET_NAME}/", "")
+
+        return jsonify(json.loads(direct_download_url(key)))
```

Comment:
Whoops missed this, should be just `jsonify(direct_download_url(key))`

Reply:
I tried that, but it broke things for me. `direct_download_url(key)` returns a JSON string, so I believe it's right to convert it to a JSON object using `json.loads()` and then convert it to a JSON response using `jsonify()`, no?

Reply:
Indeed